#!/usr/bin/env python
PACKAGE = "tiago_pcl_tutorial"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("setLeafSize_X", double_t, 0, "A double parameter",    0.01, 0,   0.1)
gen.add("setLeafSize_Y", double_t, 0, "A double parameter",    0.01, 0,   0.1)
gen.add("setLeafSize_Z", double_t, 0, "A double parameter",    0.01, 0,   0.1)
gen.add("Distance_Threshold", double_t, 0, "A double parameter",    .01, 0,   010)
gen.add("Radius_Limit", double_t, 0, "A double parameter",    .01, 0,   010)
gen.add("Optimize_Coefficients", bool_t, 0, "A bool parameter", True)

image = gen.enum([ gen.const("original_image",      int_t, 0, "original image"),
                   gen.const("filtered_image",     int_t, 1, "filtered image"),],
                     "An enum to choose origin PointCloud2")
gen.add("image", int_t, 0, "A PointCloud2 origin param which is edited via an enum", 1, 0, 3, edit_method=image)

method = gen.enum([ gen.const("Plane",      int_t, 0, "Plane"),
                   gen.const("Cylindrical",     int_t, 1, "Cylindrical"),
                   gen.const("Euclidean",     int_t, 2, "Euclidean"),],
                     "An enum to choose method")
gen.add("method", int_t, 0, "A PointCloud2 origin param which is edited via an enum", 1, 0, 3, edit_method=method)

exit(gen.generate(PACKAGE, "tiago_pcl_tutorial", "segment"))
